src/main/java/org/sagebionetworks/bridge/spring/filters/MetricsFilter.java

```diff
@@ -48,7 +55,18 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
 
         // Process the query parameters, and append them to the metrics.
         List<NameValuePair> params = URLEncodedUtils.parse(request.getQueryString(), StandardCharsets.UTF_8);
-        metrics.setQueryParams(params);
+        final Map<String, List<String>> paramsMap = new LinkedHashMap<>();
+        for (NameValuePair pair : params) {
+            String key = pair.getName();
+            String value = pair.getValue();
+            if (ALLOW_LIST.contains(key)) {
+                if (!paramsMap.containsKey(key)) {
+                    paramsMap.put(key, new LinkedList<>());
+                }
+                paramsMap.get(key).add(value);
+            }
+        }
```

Comment:
Consider using a Guava Multimap to simplify this code https://github.com/google/guava/wiki/NewCollectionTypesExplained#multimap